test/net.test.js

```diff
@@ -143,828 +144,145 @@ describe("Calculate net salary", function () {
         })
     });
 
-    describe('Check Net salary with Insurance', function () {
+    describe('Check Insurance & Taxes deduction', function () {
 
-        describe('Before 1/7/2023', function () {
+        beforeEach(function () {
+            insurancesCalcStub = sinon.stub();
+            mockedInsurance = { total: 0, insurances: [] };
 
-            test('Throw Exception when entering invalid region', function () {
-                assert.throws(function () {
-                    net(93_678_123, 0, 10, new Date("2023-06-01"));
-                }, Error, 'Invalid region entered. Please enter again! (1, 2, 3, 4)');
+            taxCalcStub = sinon.stub()
+            mockedTaxes = { totalTax: 1, rates: [] };
+        });
 
-            })
+        afterEach(function () {
+            sinon.restore();
+        })
 
-            test('It applies 10.5% of Gross for all Insurance(SI 8%, HI 1.5% , UI 1%) when Gross Income is less than or equal 20 times Base Salary (29.8m)', function () {
-                //Gross Income <= 29.8 million:
+        describe('Before 1/7/2023', function () {
```

Comment:
this describe block can be deleted AND move test() same level as beforeEach.

What to test?
**Make sure that taxes calc and insurances calc functions are called with correct Args AND their output are used correctly**